soc: silabs: s2: make order in soc Kconfig's

yishai1999 opened the pull request:
Made some order in the Kconfig's for silabs series 2 socs.

nordicjm commented on soc/silabs/silabs_s2/xg21/Kconfig.soc:
```diff
@@ -1,9 +1,15 @@
+	help
+	  Silicon Labs XG21 Series SoC and modules
+
+config SOC_SERIES_EFR32MG21
+	bool
+	select SOC_SERIES_XG21
 	help
 	  Silicon Labs EFR32MG21 (Mighty Gecko) Series MCU
```
no this is not how soc series Kconfigs work, they do not select each other, only one is selected

nordicjm replied:
then one of those is going to need to be dropped, it is described on https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#hardware-support-hierarchy

nordicjm replied:
The idea is that in future there will be no selections of these symbols (i.e. other than selecting a `SOC_MY_SOC` from a board - perhaps, or this might be done automatically too) and all SOC_* symbols will be automatically selected by the build system, so adding that does not fit in to that but would need to wait for @tejlmand to get back from vacation to comment on the suitability because the automatic build system parts is his idea

asmellby replied:
@yishai1999 Yeah, this was what I was thinking (alternatively also moving it to `Kconfig`, since it isn't used for hwmv2, but I don't know if that's important -- the docs say "This file selects the SoC family and series in the Kconfig tree and potential other SoC related Kconfig settings" [<https://docs.zephyrproject.org/latest/hardware/porting/soc_porting.html#write-kconfig-files>]).

There is prior art for this in the tree for STM32: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/st/stm32/stm32f1x/Kconfig.soc#L40. This one does it the same way as you've done here now, and for exactly the same reason -- to produce a generic symbol that is then used in a clock control driver header: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/clock_control/stm32_clock_control.h#L20

yishai1999 committed:
soc: silabs: s2: make order in soc Kconfig's

Made some order in the Kconfig's for silabs series 2 socs.
Made a distinction between silabs "generic family" (e.g. xg21) and silabs
"device family" (e.g. efr32mg21).

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

asmellby commented:
@nordicjm Do you have any additional required changes now that the "series" duplication is resolved, or can this PR be made ready to merge?

nordicjm commented:
Remember to re-request review in the top right when you want reviewers to look at a PR
